lxd/instance/drivers/driver_lxc.go

```diff
@@ -4075,6 +4070,37 @@ func (d *lxc) Update(args db.InstanceArgs, userRequested bool) error {
 		return err
 	}
 
+	// If raw.lxc changed, re-validate the config.
+	if shared.StringInSlice("raw.lxc", changedConfig) && d.expandedConfig["raw.lxc"] != "" {
+		// Get a new liblxc instance.
+		cc, err := liblxc.NewContainer(d.name, d.state.OS.LxcPath)
+		if err != nil {
+			return err
+		}
+
+		// Write the raw config.
+		f, err := ioutil.TempFile("", "lxd_config_")
+		if err != nil {
+			return err
+		}
+
+		err = shared.WriteAll(f, []byte(d.expandedConfig["raw.lxc"]))
+		f.Close()
+		defer os.Remove(f.Name())
+		if err != nil {
+			return err
+		}
+
+		// Load the raw config.
+		err = cc.LoadConfigFile(f.Name())
```

Comment:
We should have a version that takes an FD as argument. Then you can generate in-memory config files (memfd, TMobile etc.).

Reply:
*TMobile my ass: tmpfile

Reply:
I did wonder about that :)

Reply:
@stgraber so when validating raw.lxc we don't need to also have the rest of the config available, as it liblxc doesn't do cross-key contextual validation (i.e in the original posters example, specifying an IP address on a NIC that doesn't exist yet)?